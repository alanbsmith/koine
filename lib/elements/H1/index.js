import styled from 'styled-components';
import { applyStyleModifiers } from 'styled-components-modifiers';
import { rem } from 'polished';

import { fontWeights } from '../../modifiers';

const H1 = styled.h1`
  font-size: ${rem(40)};
  margin: 0;
  text-transform: capitalize;
  ${applyStyleModifiers(fontWeights)}
`;

export default H1;
