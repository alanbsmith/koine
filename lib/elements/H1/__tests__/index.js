import React from 'react';
import H1 from '../index';
import 'jest-styled-components';

import { renderWithTheme } from '__tests__/helpers';


describe('H1 Element', () => {
  it('renders correctly', () => {
    const tree = renderWithTheme(
      <H1>Header Element</H1>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('applies normalWeight modifier', () => {
    const tree = renderWithTheme(
      <H1 modifiers={['normalWeight']}>Header Element</H1>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
