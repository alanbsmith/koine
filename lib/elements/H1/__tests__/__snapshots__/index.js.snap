// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A H1 Element applies normalWeight modifier 1`] = `
<h1
  className="sc-bdVaJa iECMLI"
>
  Header Element
</h1>
`;

exports[`A H1 Element renders correctly 1`] = `
<h1
  className="sc-bdVaJa jqOjeD"
>
  Header Element
</h1>
`;
